### PR TITLE
Cleans up and DMdoc's lazarus_injector.dm.

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
title

## Why It's Good For The Game
Early returns, boolean values TRUE and FALSE instead of 1 and 0, Dmdoc.

## Changelog
Not player-facing.

### CODE REVIEW

File: code/modules/mining/equipment/lazarus_injector.dm [hunk 1]
```diff
@@ -11,52 +18,53 @@
...
+	var/loaded = TRUE
+	///Injector malf?
+	var/malfunctioning = FALSE
+	///So you can't revive boss monsters or robots with it
+	var/revive_type = SENTIENCE_ORGANIC
 
 /obj/item/lazarus_injector/afterattack(atom/target, mob/user, proximity_flag)
 	. = ..()
-	if(!loaded)
+	if((!loaded) || (!(isliving(target) && proximity_flag)))
```
Comment (Fikou) [inline code/modules/mining/equipment/lazarus_injector.dm:30]: why are you loading so much brackets in

Comment (interestingusernam3) [reply code/modules/mining/equipment/lazarus_injector.dm:30]: To make the code work.

File: code/modules/mining/equipment/lazarus_injector.dm [hunk 2]
```diff
@@ -11,52 +18,53 @@
...
-				icon_state = "lazarus_empty"
-				return
-			else
-				to_chat(user, span_info("[src] is only effective on the dead."))
-				return
+
+	M.faction = list("neutral")
+	M.revive(full_heal = TRUE, admin_revive = TRUE)
+	if(ishostile(target))
+		var/mob/living/simple_animal/hostile/H = M
```
Comment (Rohesie) [inline code/modules/mining/equipment/lazarus_injector.dm:47]: Can you give these `M` and `H` variables better names?

File: code/modules/mining/equipment/lazarus_injector.dm [hunk 3]
```diff
@@ -11,52 +18,53 @@
...
 /obj/item/lazarus_injector/afterattack(atom/target, mob/user, proximity_flag)
 	. = ..()
-	if(!loaded)
+	if(!loaded || !(isliving(target) && proximity_flag) )
+		return
+	if(!isanimal(target))
+		to_chat(user, span_info("[src] is only effective on lesser beings."))
+		return
+
+	var/mob/living/simple_animal/M = target
```
Comment (Kylerace) [inline code/modules/mining/equipment/lazarus_injector.dm:36]: descriptive var name please

File: code/modules/mining/equipment/lazarus_injector.dm [hunk 4]
```diff
@@ -11,52 +18,53 @@
...
-				to_chat(user, span_info("[src] is only effective on the dead."))
-				return
+
+	M.faction = list("neutral")
+	M.revive(full_heal = TRUE, admin_revive = TRUE)
+	if(ishostile(target))
+		var/mob/living/simple_animal/hostile/H = M
+		if(malfunctioning)
+			H.faction |= list("lazarus", "[REF(user)]")
+			H.robust_searching = 1
```
Comment (Kylerace) [inline code/modules/mining/equipment/lazarus_injector.dm:50]:
```suggestion
			H.robust_searching = TRUE
```

File: code/modules/mining/equipment/lazarus_injector.dm [hunk 5]
```diff
@@ -11,52 +18,53 @@
...
+			H.friends += user
+			H.attack_same = 1
```
Comment (Kylerace) [inline code/modules/mining/equipment/lazarus_injector.dm:52]:
```suggestion
			H.attack_same = TRUE
```

File: code/modules/mining/equipment/lazarus_injector.dm [hunk 6]
```diff
@@ -11,52 +18,53 @@
...
+			log_game("[key_name(user)] has revived hostile mob [key_name(target)] with a malfunctioning lazarus injector")
 		else
-			to_chat(user, span_info("[src] is only effective on lesser beings."))
-			return
+			H.attack_same = 0
```
Comment (Kylerace) [inline code/modules/mining/equipment/lazarus_injector.dm:55]:
```suggestion
			H.attack_same = FALSE
```

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/mining/equipment/lazarus_injector.dm (hunk 2 above)
Comment (interestingusernam3) [reply code/modules/mining/equipment/lazarus_injector.dm]: done

File: code/modules/mining/equipment/lazarus_injector.dm (hunk 3 above)
Comment (interestingusernam3) [reply code/modules/mining/equipment/lazarus_injector.dm]: done

File: code/modules/mining/equipment/lazarus_injector.dm (hunk 4 above)
Comment (interestingusernam3) [reply code/modules/mining/equipment/lazarus_injector.dm]: done

File: code/modules/mining/equipment/lazarus_injector.dm (hunk 5 above)
Comment (interestingusernam3) [reply code/modules/mining/equipment/lazarus_injector.dm]: done

File: code/modules/mining/equipment/lazarus_injector.dm (hunk 6 above)
Comment (interestingusernam3) [reply code/modules/mining/equipment/lazarus_injector.dm]: done

**Pull request merged**